doc/adr/0003-dag-l1-responsibility-and-l0-interconnection.md

```diff
@@ -0,0 +1,30 @@
+# 3. DAG L1 responsibility and L0 interconnection
+
+Date: 2022-01-05
+
+## Status
+
+Proposed
+
+## Context
+
+DAG L1 responsibility and communication between DAG L1 and L0 needs to be
+specified.
```

Comment:
L0 sends relevent Snapshot data as a metamorphism/stream (using same algebra/coalgebra in reverse [anna cata)])
L1 sends blocks as a hylomorphism (using algebra/coalgebra as specified [cata/anna])

Reply:
I'd argue with that. L1 knows the "connection point" to the Layer 0. It can be a load balancer or any single peer (data should be gossiped across the whole layer anyway). While L0 doesn't know the L1 at all. Doesn't know the L1 peers etc. So what L0 can do is to share the stream of data (let's simplify it to an endpoint) so then L1 can subscribe to this and watch for changes or new events/data (pull-based instead of push-based).

Reply:
I think that it's wrong to say that L0 sends relevant Snapshot data to L1, because L0 doesn't know the L1 peers.

Reply:
and actually it implies that the L0 should work without any L1. L1 is a deliverer of data but if there are no data, L0 should work anyway.

Reply:
Well for the DAG L0, in order to validate blocks, it will need to be able to validate the facilitators/edges for each block. So it will need the information about a subset of L1 that are selected for active consensus. We can reuse the same process cycling L0 nodes from pending to active as L1. But that should be it